### PR TITLE
Fixing linting issues for generated containers

### PR DESCRIPTION
**Pull request by Nuhvi**
### Problem
Generated Containers with reducers, sagas and/or tests, produce linting issues:
- Reducers' and Sagas' keys need to be manually imported to `types\index.d.ts`.
- Generated `tests/.index` uses the extension .ts instead of .tsx
-  `Jsx-wrap-multiline` rule produce false positives in `tests/.index.tsx`

### Changes
- Add special comments in `app/types/index.d.ts` for the plop #modify action to find.
- Add two actions in `internals/generators/container/index.js` to automate adding new reducers key to RootState.
- Remove previous warning comments from the existing files and templates.
- Change the  `internals/generators/container/index.js` to create `tests/index.test.tsx` instead of `tests/index.test.ts`.
- Change `internals/generators/container/test.tsx.hbs` to use 'utils/history' instead of `browserHistory`.
- Disable `Jsx-wrap-multiline` in tslint.json and remove redundant comments everywhere else.

### Result and limitations
Currently, generating either Components or Container using any set of options will pass tslint.
I can't say the same about Languages, and I still couldn't fix all issues with generate-templates-for-linting.js yet.
If you would like to help with that please let me know, I am willing to put the work but I am very inexperienced with the current setup and some guidance might speed things up.

Thanks for your patience.

### CODE REVIEW

**Comment by Can-Sahin** [comment]
I just pushed some changes to `dev` that fixes some stuff.  I am not able to get `travis-ci` running because I moved this repository recently and name changes kinda messed things up. 

Can you pull from dev again and see if any conflicts. And run `npm run test` and `node ./internals/scripts/generate-templates-for-linting` to see everything works fine.

Then I can merge. Now it says lot of commits since your version of `dev` was far behind the `origin master`. (I merged master into dev after this PR ;) )

**Comment by Nuhvi** [comment]
I will do that tonight! thanks!

**Comment by Nuhvi** [comment]
https://travis-ci.org/Nazeh/react-boilerplate-typescript/builds/652592673

@Can-Sahin It passed, the generate script doesn't clean the types/index.d.ts, so I just configured Travis to stash the changes before testing, as this code is not used by users, this should be enough I guess.

TODO

Add a remove script that will clean containers and their side effects in types.
let me know if that is acceptable before I implement it.

File: internals/templates/types/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { Reducer, Store } from 'redux';
 import { RouterState } from 'connected-react-router';
 import { ContainerState as LanguageProviderState } from 'containers/LanguageProvider/types';
+// [IMPORT NEW CONTAINERSTATE ABOVE] < Needed for generating continers seamlessly
```
Comment (Can-Sahin) [inline internals/templates/types/index.d.ts]: typo -> containers

File: internals/templates/types/index.d.ts [hunk 2]
```diff
@@ -23,6 +24,8 @@ export interface InjectSagaParams {
 export interface ApplicationRootState {
   readonly router: RouterState;
   readonly language: LanguageProviderState;
+  // [INSERT NEW REDUCER KEY ABOVE] < Needed for generating continers seamlessly
```
Comment (Can-Sahin) [inline internals/templates/types/index.d.ts]: typo -> containers

File: jest.config.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 module.exports = {
   preset: 'ts-jest/presets/js-with-babel',
+  testPathIgnorePatterns: ['RbGen*'],
```
Comment (Can-Sahin) [inline jest.config.js]: why is this needed ? Tests are never run with generated components

**Comment by Can-Sahin** [comment]
That cannot be a TODO. Otherwise every time I run that function I have to manually discard changes. There are cleanup functions just for that ;) You can add another `backup` like language generator do and use `restoreModifiedFile` to restore the original `index.d.ts` file back. Just have a look at how language generator does that in that script.  

Looks good otherwise. Thanks for the contribution.

File: jest.config.js (hunk 3 above)
Comment (Nuhvi) [reply jest.config.js]: Yes, you are right, I added that because when the script failed in the middle, it didn't clean up the generated components, I discovered later that normally they are removed.

I will remove that.

**Comment by Nuhvi** [comment]
I meant that if a user created a container, then decided to remove it, the keys in types/index.d.ts will still need to be removed, and more importantly, the import of ContainerState will fail.
User can remove them manually but that will risk missing with the special comment:
// [IMPORT NEW CONTAINERSTATE ABOVE] < Needed for generating containers seamlessly

So I was thinking of a special script to avoid manually touching types/index.d.ts.. but I am not sure it is worth it especially that it will deviate from the experience of the original boilerplate.

**Comment by Can-Sahin** [comment]
`git` command in `travis.yml` is not a good idea. It probably won't work. You aren't in your local environment there. Its in a build server and it is surely bad to run git commands on a build server. 

Users can fix their own removed containers for sure. Test code should delete its own as well.

**Comment by Nuhvi** [comment]
It does work at least on my Travis builds:
https://travis-ci.org/Nazeh/react-boilerplate-typescript/builds/652647946?utm_source=github_status&utm_medium=notification

If it is still unacceptable, I will try to work on it tomorrow.

**Comment by Can-Sahin** [comment]
I would fix it definitely. Better not to leave loose ends. I can do it as well later. Using already implemented functions would be enough like I said above. Same logic is already handled in language generator tests.

**Comment by Nazeh** [comment]
I am not sure if the existing function is useful, but I am almost done. I will push my commit in minutes.

**Comment by Nuhvi** [comment]
[Done](https://travis-ci.org/Nazeh/react-boilerplate-typescript/builds/652714042?utm_source=github_status&utm_medium=notification)
